primitives/merkle-patricia-trie/src/error.rs

```diff
@@ -22,35 +22,26 @@ extern crate alloc;
 #[cfg(not(feature = "std"))]
 use alloc::{format, string::String};
 
+use core::fmt;
 use rlp::DecoderError;
 #[cfg(not(feature = "std"))]
 use sp_std::borrow::ToOwned;
-use sp_std::fmt;
 
-#[derive(Debug)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub enum TrieError {
 	DB(String),
 	Decoder(DecoderError),
 	InvalidData,
 	InvalidStateRoot,
 	InvalidProof,
 }
-
 impl fmt::Display for TrieError {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		let printable = match *self {
-			TrieError::DB(ref err) => format!("trie error: {:?}", err),
-			TrieError::Decoder(ref err) => format!("trie error: {:?}", err),
-			TrieError::InvalidData => "trie error: invalid data".to_owned(),
-			TrieError::InvalidStateRoot => "trie error: invalid state root".to_owned(),
-			TrieError::InvalidProof => "trie error: invalid proof".to_owned(),
-		};
-		write!(f, "{}", printable)
+		fmt::Debug::fmt(&self, f)
```

Comment:
In this way, the output of `TrieError` will be the same with `#[derive(Debug)]`.How about removing `impl fmt::Display for TrieError {...}`?

Reply:
Out of scope of this PR.

There is an incoming refactor about `merkle-patricia-trie`.